From 900eff6ae477b2674b8504d5420d565ca09ac813 Mon Sep 17 00:00:00 2001
From: Robert Schiele <rschiele@gmail.com>
Date: Fri, 24 Jun 2022 16:50:54 +0200
Subject: [PATCH] take contrib blocklist into account for doc generation

If some contrib module is blocked for building also block it in
documentation generation since this will fail otherwise.

Additionally this fixes a shift operation that is needed
unconditionally when parsing the script parameters since otherwise
the first parameter is used for the second parameter as well if it
turns out to be empty.
---
 doc/manual/generate-texinfo.lisp | 9 +++++----
 doc/manual/make-tempfiles.sh     | 8 +++++---
 2 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/doc/manual/generate-texinfo.lisp b/doc/manual/generate-texinfo.lisp
index 83451754d..44e5edca8 100644
--- a/doc/manual/generate-texinfo.lisp
+++ b/doc/manual/generate-texinfo.lisp
@@ -32,8 +32,9 @@
         finally (return result)))
 
 (defun generate-docstrings-texinfo (runtime
-                                    &key (docstring-directory "docstrings/"))
-  (let* ((contribs (sort (documented-contribs) #'string< :key #'car))
+                                    &key (docstring-directory "docstrings/")
+                                   (blocklist '()))
+  (let* ((contribs (sort (documented-contribs :exclude (append '("asdf") blocklist)) #'string< :key #'car))
          (packages (sort (append *documented-packages*
                                  (map 'list #'cdr contribs))
                          #'string<)))
@@ -92,10 +93,10 @@
 
 ;;;; Entry point
 
-(destructuring-bind (program runtime docstring-directory) *posix-argv*
+(destructuring-bind (program runtime docstring-directory blocklist) *posix-argv*
   (declare (ignore program))
   (generate-docstrings-texinfo
-   runtime :docstring-directory docstring-directory)
+   runtime :docstring-directory docstring-directory :blocklist (uiop:split-string blocklist))
 
   (expand-variables)
   (generate-external-format-texinfo))
diff --git a/doc/manual/make-tempfiles.sh b/doc/manual/make-tempfiles.sh
index bf36ab1ec..faf1bbbf9 100644
--- a/doc/manual/make-tempfiles.sh
+++ b/doc/manual/make-tempfiles.sh
@@ -26,18 +26,20 @@ if [ -z "$1" ] ; then
     else
         SBCLRUNTIME="`command -v sbcl`"
     fi
+    . $SBCL_TOP/output/build-config
 else
     SBCLRUNTIME="$1"
-    shift
+    SBCL_CONTRIB_BLOCKLIST=
 fi
+shift
 
 if [ -z "$1" ] ; then
     DOCSTRINGDIR="${DOCSTRINGDIR:-docstrings/}"
 else
     DOCSTRINGDIR="$1"
-    shift
 fi
+shift
 
 ${SBCLRUNTIME}                                                          \
     --noinform --no-sysinit --no-userinit --noprint --disable-debugger  \
-    --script generate-texinfo.lisp "${SBCLRUNTIME}" "${DOCSTRINGDIR}"
+    --script generate-texinfo.lisp "${SBCLRUNTIME}" "${DOCSTRINGDIR}" "${SBCL_CONTRIB_BLOCKLIST}"
-- 
2.36.1

